src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/EditingTargetCodeExecution.java

```diff
@@ -257,6 +229,53 @@ private void executeRange(Range range, String functionWrapper, boolean onlyUseCo
                                   prefs_.focusConsoleAfterExec().getValue()));
    }
    
+   public void executeBehavior(String executionBehavior)
+   {
+      Range range = getRangeFromBehavior(executionBehavior);
+      executeRange(range, null, false);
+      moveCursorAfterExecution(range);
+   }
+   
+   private Range getRangeFromBehavior(String executionBehavior)
+   {
+      Range range;
+      
+      // by default the range can encompass the whole document
+      int startRowLimit = 1;
```

Comment:
I think so! I copied the 1 from the previous implementation, which apparently was also wrong since rows are 0-based. I've fixed this up.